pkg/awsutils/awsutils.go

```
@@ -0,0 +1,36 @@
+package awsutils
+
+/**
+ * Panther is a Cloud-Native SIEM for the Modern Security Team.
+ * Copyright (C) 2020 Panther Labs Inc
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+import "github.com/aws/aws-sdk-go/aws/awserr"
+
+// Method returns true if the provided error is an AWS error with any
+// of the given codes.
+func IsAnyError(err error, codes ...string) bool {
+	awserror, ok := err.(awserr.Error)
```

Careful: this will *not* work if the error is wrapped with the `errors` package. This led to some serious bugs: #1517 

To be safe, use `errors.As` instead of checking the type explicitly:

```go
    var awserror
    if errors.As(err, &awserror) {
```

This function will work whether the error is wrapped or not, or if it's nil. So it's safer overall

---

I know it adds a dependency to this library that isn't technically needed, but we use the `errors` package throughout most of our code - I don't want a helper library that will silently fail for a large class of our own errors

---

Good suggestion.

---

I was thinking it might make sense to have two methods : `IsAnyError` and `ContainsAnyError`. The first checks if the `error` provided matches any of the provided codes. The second checks if any of the errors in the chain match the provided codes. I was thinking there might be a case where we want to distinguish between the two scenarios. 
Do you think it makes sense? Or will we end up using the `Contains` all the time? 
I made a comment about this [here](https://github.com/panther-labs/panther/pull/1517#discussion_r486823329)

---

Most of the time, we're checking for a single AWS error code. But sometimes that's not sufficient and we have to instead check the error message. Very rarely, we check for more than one error code. Personally I think the helper function is overkill for this, but doesn't really matter to me either way. I was just want to make sure this:

```go
if awsErr, ok := err.(awserr.Error)
```

doesn't appear anywhere in the codebase, since it will not work with the `errors` package